include/ginkgo/core/solver/solver_base.hpp

```diff
@@ -50,6 +51,142 @@ namespace gko {
 namespace solver {
 
 
+/**
+ * Give a initial guess about the input of the apply method.
+ */
+enum class initial_guess {
```

Comment:
nit: This enum doesn't describe an initial guess, it describes how an initial guess is supposed to be chosen
```suggestion
enum class initial_guess_mode {
```